inc/aiosp_common.php

```diff
@@ -182,4 +189,255 @@ public static function make_xml_safe( $tag, $value ) {
+	 *
+	 * @see aiosp_common::set_transient_url_postids()
+	 * @see get_transient()
+	 * @link https://developer.wordpress.org/reference/functions/get_transient/
+	 * @see wpdb::get_results()
+	 * @link https://developer.wordpress.org/reference/classes/wpdb/get_results/
+	 * @see wp_list_pluck()
+	 * @link https://developer.wordpress.org/reference/functions/wp_list_pluck/
+	 * @see wp_upload_dir()
+	 * @link https://developer.wordpress.org/reference/functions/wp_upload_dir/
+	 *
+	 * @since 2.9.2
+	 *
+	 * @param string $url Full image URL.
+	 * @return int
+	 */
+	public static function attachment_url_to_postid( $url ) {
+		global $wpdb;
+		static $results_1;
+		static $results_2;
+
+		$id = 0;
+		$url_md5 = md5( $url );
+
+		// Gets the URL => PostIDs array.
+		// If static variable is still empty, load transient data.
+		if ( is_null( self::$attachment_url_postids ) ) {
+			if ( is_multisite() ) {
+				self::$attachment_url_postids = get_site_transient( 'aioseop_multisite_attachment_url_postids' );
+			} else {
+				self::$attachment_url_postids = get_transient( 'aioseop_attachment_url_postids' );
+			}
+
+			// If no transient data, set as (default) empty array.
+			if ( false === self::$attachment_url_postids ) {
+				self::$attachment_url_postids = array();
+			}
+		}
+
+		// Search for URL and get ID.
+		if ( isset( self::$attachment_url_postids[ $url_md5 ] ) ) {
+			// If static is already loaded and has URL, then return the URL's Post ID.
+			$id = intval( self::$attachment_url_postids[ $url_md5 ] );
+		} else {
+			// Check to make sure Image URL is not outside the website.
+			$uploads_dir = wp_upload_dir();
+			if ( false !== strpos( $url, $uploads_dir['baseurl'] . '/' ) ) {
+				// Results_1 query looks for URLs with the original guid that is uncropped and unedited.
+				if ( is_null( $results_1 ) ) {
+					$results_1 = aiosp_common::attachment_url_to_postid_query_1();
+				}
+
+				if ( isset( $results_1[ $url_md5 ] ) ) {
+					$id = intval( $results_1[ $url_md5 ] );
+				}
+
+				// TODO Add setting to enable; this is TOO MEMORY INTENSE which could result in 1 or more crashes,
+				// TODO however some may still need custom image URLs.
+				// TODO NOTE: Transient data does prevent continual crashes.
+//				else {
+//					// Results_2 query looks for the URL that is cropped and edited. This searches JSON strings
+//					// and returns the original attachment ID (there is no custom attachment IDs).
+//
+//					if ( is_null( $results_2 ) ) {
+//						$results_2 = aiosp_common::attachment_url_to_postid_query_2();
+//					}
+//
+//					if ( isset( $results_2[ $url_md5 ] ) ) {
+//						$id = intval( $results_2[ $url_md5 ] );
+//					}
+//				}
+			}
+
+			self::$attachment_url_postids[ $url_md5 ] = $id;
+
+			/**
+			 * Sets the transient data at the last hook instead at every call.
+			 *
+			 * @see aiosp_common::set_transient_url_postids()
+			 */
+			add_action( 'shutdown', 'aiosp_common::set_transient_url_postids' );
```

Comment:
@EkoJR I suggest using `add_action( 'shutdown', array( 'aiosp_common', 'set_transient_url_postids' ) )` instead

Reply:
I could of swore that had issues with static functions, but I guess I forgot to change it back at some point.